multicluster/controllers/multicluster/serviceexport_controller.go

```diff
@@ -138,15 +141,17 @@ func (r *ServiceExportReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	epResExportName := getResourceExportName(r.localClusterID, req, "endpoints")
 
 	cleanup := func() error {
+		// There is no side effect to call deletion directly, so leave it to the caller
```

Comment:
I think this is not the reason. The reason is when controller restarts, the Service is not in cache, but it is still possible we need to remove ResourceExports.

Reply:
Updated